lib/clustering/mean_shift.h

```diff
@@ -9,19 +9,39 @@
 #include "clustering.h"
 #include <vector>
 #include "generic/cluster.h"
+#include <Eigen/Core>
 
 namespace MouseTrack {
 
 class MeanShift: public Clustering {
 public:
 
   /// k is the number of desired clusters
-  MeanShift(const int k);
+  MeanShift(double window_size);
   /// Splits a point cloud into k clusters randomly
   std::vector<Cluster> operator()(const PointCloud& cloud) const;
 
 private:
-    const int _k;
+  // when two peaks are closer than this, they are merged.
+  const double MERGE_THRESHOLD = 0.0001;
+
+  // Convergence is decided if modes haven't moved more than this in total
+  const double CONVERGENCE_THRESHOLD = 0.1;
+
+  // If algorithm hasn't converged after this amount of iterations, we abort.
+  const int MAX_ITERATIONS = 100;
+
+  // window size parameter for mean shift algorithm
+  double _window_size;
+
+  // contains the means of all the mean shift clusters
+  PointCloud means;
```

Comment:
dead code? since the method is "const", this should probably not exist.

Reply:
You're right, forgot to delete. My bad.